vignettes/howtogenefilter.Rmd

```diff
@@ -0,0 +1,140 @@
+---
+title: "Using the genefilter function to filter genes from a microarray dataset"
+author:
+- name: "Khadijah Amusat"
+  affiliation: "Vignette translation from Sweave to R Markdown / HTML"
+date: "`r format(Sys.time(), '%B %d , %Y')`"
+output:
+    BiocStyle::html_document:
+    number_sections: true
+    toc: true
+    toc_depth: 4
+package: genefilter
+vignette: >
+  %\VignetteIndexEntry{Using the genefilter function to filter genes from a microarray}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+
+# Introduction {#introduction .unnumbered}
+
+The `r Biocpkg("genefilter")` package can be used to filter (select) genes from a microarray dataset according to a variety of different filtering mechanisms. Here, we will consider the example dataset in the `sample.ExpressionSet` example from the `r Biocpkg("Biobase")` package. This experiment has 26 samples, and there are 500 genes and 3 covariates. The covariates are named `sex`, `type` and `score`. The first two have two levels and the last one is continuous.
+
+```{r, closeg, message = FALSE}
+library(Biobase) 
+library(genefilter)
+data(sample.ExpressionSet)
+varLabels(sample.ExpressionSet)
+table(sample.ExpressionSet$sex)
+table(sample.ExpressionSet$type)
+data(sample.ExpressionSet)
+```
+
+ One dichotomy that can be of interest for subsequent analyses is whether the filter is *specific* or *non-specific*. Here, specific means that we are filtering with reference to sample metadata, for example, `type`. For example, if we want to select genes that are differentially expressed in the two groups defined by `type`, that is a specific filter. If on the other hand we want to select genes that are expressed in more than 5 samples, that is an example of a non--specific filter.
+
+ First, let us see how to perform a non--specific filter. Suppose we want to select genes that have an expression measure above 200 in at least 5 samples. To do that we use the function `kOverA`.
+
+There are three steps that must be performed.
+
+1.  Create function(s) implementing the filtering criteria.
+
+2.  Assemble it (them) into a (combined) filtering function.
+
+3.  Apply the filtering function to the expression matrix.
+
+```{r, message=FALSE}
+f1 <- kOverA(5, 200)
+ffun <- filterfun(f1)
+wh1 <- genefilter(exprs(sample.ExpressionSet), ffun)
+sum(wh1)
+```
+
+Here `f1` is a function that implies our "expression measure above 200 in at least 5 samples" criterion, the function `ffun` is the filtering function (which in this case consists of only one criterion), and we apply it using `genefilter`. There were  `r {sum(wh1)}` genes that satisfied the criterion and passed the filter.
```

Comment:
Remove the curly braces `{}` as they aren't necessary for inline code. Do this for other instances in this vignette.